sql/parse/parse.go

```diff
@@ -4346,7 +4346,11 @@ func convertVal(ctx *sql.Context, v *sqlparser.SQLVal) (sql.Expression, error) {
 		// use the value as string format to keep precision and scale as defined for DECIMAL data type to avoid rounded up float64 value
 		if ps := strings.Split(string(v.Val), "."); len(ps) == 2 {
 			ogVal := string(v.Val)
-			floatVal := strconv.FormatFloat(val, 'f', -1, 64)
+			var fmtStr byte = 'f'
+			if strings.Contains(ogVal, "e") {
```

Comment:
I haven't spent a lot of time researching this, but would be really nice if we could use a standardized library, or something with a bit more guard rails. https://pkg.go.dev/math/big#ParseFloat maybe? Or maybe a regex to verify scientific notation, if that isn't super hard to do